docs/dev-env.md

```diff
@@ -86,6 +86,30 @@ project they will automatically set up the environment for you, without starting
 new shells. This way you can preserve your shell, and your settings while using
 `nix develop`-like shell automatically.
 
+## Cachix binary cache
+
+If you run `nix develop` immediately after following the instructions above, nix
+will proceed to build all the dependencies that _aren't_ part of nixpkgs (and
+so aren't cached by <https://cache.nixos.org>) from scratch. This can take a
+long time. Fedimint uses [Cachix](https://www.cachix.org/) to provide binary
+caches of all the project's dependencies. It is possible to trust this cache,
+and download pre-built binaries instead.
+
+To trust <https://fedimint.cachix.org>, run the following command in your
+terminal and respond to the prompts.
+
+```
+nix develop .#bootstrap -c cachix use fedimint
+```
+
+This invokes the Cachix CLI and updates your nix configuration. It adds the
+Cachix cache as a substituter and trusts its public key. See the nix
+documentation on [substituters](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-substituters)
+and [trusted public keys](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-trusted-public-keys)
+for more information. 
+
+Now you can run `nix develop` and benefit from the binary cache.
```

Comment:
`nix develop` should ask on first start if you want to trust our cachix repo as well.

Reply:
Yeah agree. Not sure if that's possible. 

Did a little digging, and it's possible to "extend" nix.conf on a flake-by-flake basis using the `nixConfig` output.

I see this already exists:
https://github.com/fedimint/fedimint/blob/85f4e4944e582f9e6e4ca91bea287c5ef9dde26b/flake.nix#L377-L380

BUT, the catch is that this only has an effect when the user sets the [`accept-flake-config`](https://nixos.org/manual/nix/stable/command-ref/conf-file#conf-accept-flake-config) global setting. 

So, maybe the solution is to encourage the user to turn on "accept-flake-config" in the docs.

Reply:
> Yeah agree. Not sure if that's possible.

@emilioziniades I mean - that's already happening, or is at least supposed to.

https://github.com/fedimint/fedimint/blob/8d88f30ab25f55e37cbe4d46572cba2dfdeda434/flake.nix#L377

is responsible for it, and it always worked for me (thought I already answered it and saved my choice so I don't see it anymore).

Reply:
> BUT, the catch is that this only has an effect when the user sets the [`accept-flake-config`](https://nixos.org/manual/nix/stable/command-ref/conf-file#conf-accept-flake-config) global setting.

Nope. I don't have that set, and would not enable it globally. I personally don't want to trust any binary caches on my personal devices. I still get the prompt asking.

Reply:
> Nope. I don't have that set, and would not enable it globally. I personally don't want to trust any binary caches on my personal devices. I still get the prompt asking.

Makes complete sense.

> @emilioziniades I mean - that's already happening, or is at least supposed to.

Ah apologies, I misunderstood you.

When I run `nix develop` for the first time, I get this warning
```
warning: ignoring untrusted substituter 'https://fedimint.cachix.org', you are not a trusted user.
```
And then it proceeds to build everything from source.

This is what my nix.conf (using nix-darwin) looks like:

```
# WARNING: this file is generated from the nix.* options in
# your nix-darwin configuration. Do not edit it!
allowed-users = *
auto-optimise-store = false
build-users-group = nixbld
builders = 
cores = 0
experimental-features = nix-command flakes
extra-sandbox-paths = 
max-jobs = auto
require-sigs = true
sandbox = false
sandbox-fallback = false
trusted-substituters = 
trusted-users = root
substituters = https://cache.nixos.org https://cache.nixos.org/
trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
```

But when I add my user to trusted-users, then it uses cachix without any issues.

Have you added your user to `trusted-users`?

And, do you know if there are any security implications to doing so?

Reply:
>  you are not a trusted user.


OOooooooh. I see. Nix build daemon will only trust users in `trusted-users` with substituting binary caches. That's why it won't even display the prompt for users that don't have their unix user added there.

I didn't think of that.

Reply:
So yeah, I guess it's worth documenting that the user needs to be in a `trusted-users` config for binary caching to work.

Reply:
> So yeah, I guess it's worth documenting that the user needs to be in a `trusted-users` config for binary caching to work.

Ok cool, I've modified the docs change to reflect what we've discussed here - would you mind taking another look?